charts/kubewarden-controller/templates/deployment.yaml

```diff
@@ -46,21 +48,55 @@ spec:
         - --leader-elect
         - --deployments-namespace={{ .Release.Namespace }}
         - --webhook-service-name={{ include "kubewarden-controller.fullname" . }}-webhook-service
-       {{- if .Values.telemetry.metrics.enabled }}
+        - --always-accept-admission-reviews-on-deployments-namespace
+        - --zap-log-level={{ .Values.logLevel }}
+       {{- if or .Values.telemetry.metrics .Values.telemetry.tracing }}
+       {{- if eq .Values.telemetry.mode "sidecar" }}
+        - --enable-otel-sidecar
+       {{- end }}
+       {{- if .Values.telemetry.metrics }}
         - --enable-metrics
        {{- end }}
-       {{- if .Values.telemetry.tracing.enabled }}
+       {{- if .Values.telemetry.tracing }}
         - --enable-tracing
        {{- end }}
-        - --always-accept-admission-reviews-on-deployments-namespace
-        - --zap-log-level={{ .Values.logLevel }}
+        {{- if and (not .Values.telemetry.custom.insecure) .Values.telemetry.custom.otelCollectorCertificateSecret }}
+        - --opentelemetry-certificate-secret={{ .Values.telemetry.custom.otelCollectorCertificateSecret }}
+        {{- end }}
+        {{- if and (not .Values.telemetry.custom.insecure) .Values.telemetry.custom.otelCollectorClientCertificateSecret }}
+        - --opentelemetry-client-certificate-secret={{ .Values.telemetry.custom.otelCollectorClientCertificateSecret }}
+        {{- end }}
+       {{- end }}
         command:
         - /manager
-        {{- if .Values.telemetry.metrics.enabled }}
         env:
+        {{- if and .Values.telemetry.metrics (eq .Values.telemetry.mode "sidecar") }}
           - name: KUBEWARDEN_POLICY_SERVER_SERVICES_METRICS_PORT
-            value: "{{ .Values.telemetry.metrics.port | default 8080 }}"
+            value: "{{ .Values.telemetry.sidecar.metrics.port | default 8080 }}"
+          - name: OTEL_EXPORTER_OTLP_ENDPOINT
+            value: "https://localhost:4317"
+          - name: OTEL_EXPORTER_OTLP_INSECURE
+            value: "true"
```

Comment:
Is this change correct, to default to insecure in sidecar?
We used to default to false (by not giving it value)
https://github.com/kubewarden/helm-charts/pull/581/files#diff-b163260f07e6fcac4ab913311793d4fe95c6cb21193ca0656fa2a022c1645449L27

Reply:
Good catch.

Reply:
Actually, the `true` values is the right one. This env var configure the Otel exporter configured inside the controller. Which sends data to the sidecar always using a insecure communication. The configuration you pointed out is the TLS configuration for the Otel collector to Jaeger. 

We can confirm in the old code of the controller where we always use [insecure](https://github.com/kubewarden/kubewarden-controller/blob/4216ecd2410d56588f5823dd46e621d08820fa7c/internal/metrics/metrics.go#L29) communication with the Otel sidecar

Reply:
You are right. Argh, sorry for the misdirection there.